packages/smooth_app/lib/cards/data_cards/image_upload_card.dart

```diff
@@ -11,15 +11,21 @@ import 'package:openfoodfacts/model/Product.dart';
 import 'package:openfoodfacts/openfoodfacts.dart';
 import 'package:openfoodfacts/utils/LanguageHelper.dart';
 
+import 'package:smooth_app/pages/product/product_image_page.dart';
+
 class ImageUploadCard extends StatefulWidget {
   const ImageUploadCard({
     this.product,
     this.imageField,
+    this.imageUrl,
```

Comment:
I think you'd be better off with 2 distinct classes: one for the upload (this one without your changes) and one for known image urls (only the changes) (perhaps `ImageUrlCard`?)

Reply:
I tried to do that after reading your comment, but then I would have to introduce another class ImageOrImageUploadCard as well, that would return one or the other. And also I wanted the ImageUploadCard to turn into an ImageCard as soon as the user has taken a picture.

And even if we already have a picture, we'll probably display a button to allow uploading a new picture. So the functionality is basically the same.

So I ended up keeping only one class.

Reply:
I understand the reasons behind your final choice but I'm not 100% convinced. We'll see that later.
But what puzzles me is that `imageUrl` (for instance) can be `null`: it should be `final String? imageUrl`, shouldn't it?

Reply:
that's right, it can be null. But if I put "final String? imageUrl", then I get this analysis error:

error: This requires the 'non-nullable' language feature to be enabled. (experiment_not_enabled at [smooth_app] lib/cards/data_cards/image_upload_card.dart:27)

we have this in pubspec.yaml:

environment:
  sdk: ">=2.10.4 <3.0.0"

and we need to set the minimum 2.12.0 to enable null safety

Reply:
My bad: it's openfoodfacts-dart that migrated to null-safety, not smoothie - that is, not yet :)